src/Parser/ClassParser.php

```diff
@@ -0,0 +1,165 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace ApiPlatform\PDGBundle\Parser;
+
+final class ClassParser extends AbstractParser
+{
+    public function __construct(private readonly \ReflectionClass $reflection)
+    {
+    }
+
+    public function hasTag(string $searchedTag): bool
+    {
+        // class has no doc (only search for class doc without inheritance)
+        if (!$this->getReflection()->getDocComment()) {
+            return false;
+        }
+
+        foreach ($this->getPhpDoc()->getTags() as $tag) {
+            if ($searchedTag === $tag->name) {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    public function getParentClass(): self|false
+    {
+        if ($parentClass = $this->getReflection()->getParentClass()) {
+            return new self($parentClass);
+        }
+
+        return false;
+    }
+
+    public function getInterfaces(): array
+    {
+        return array_map(fn (\ReflectionClass $class) => new self($class), $this->getReflection()->getInterfaces());
+    }
+
+    public function getDocComment(): string|false
+    {
+        // class has no doc
+        if (!$docComment = parent::getDocComment()) {
+            return $docComment;
+        }
+
+        // import and replace "inheritdoc" (except for traits)
+        if (!$this->getReflection()->isTrait() && str_contains($docComment, '@inheritdoc')) {
+            // import docComment from parent class, not from interfaces
+            if (false !== ($parentClass = $this->getParentClass()) && '' !== ($parentDocComment = $parentClass->getDocComment())) {
+                $docComment = preg_replace('/{?@inheritdoc}?/', $parentDocComment, $docComment);
+            }
+        }
+
+        // todo check "@see" tags to import absolute namespace if available
+
+        return $docComment;
+    }
+
+    public function getCodeSelector(): array
+    {
+        $codeSelector = [];
+        $blocks = preg_split('/(\[codeSelector\][\s\S\w\n]*?\[\/codeSelector\])/', $this->getReflection()->getDocComment(), 0, \PREG_SPLIT_DELIM_CAPTURE);
+        foreach ($blocks as $i => $block) {
+            if (str_contains($block, 'codeSelector')) {
+                if (false !== preg_match_all('/```(\w+)/', $block, $languages) && $languages) {
```

Comment:
not really related to this PR but we could merge this ifs and also `continue` to avoid the nested lines